core/src/test/java/com/sequenceiq/cloudbreak/service/image/CachedImageCatalogWrapperProviderTest.java

```diff
@@ -0,0 +1,334 @@
+
+    private static final String CB_IMAGE_CATALOG_NULL_FIELD_JSON = "cb-image-catalog-null-field.json";
+
+    private static final String CB_IMAGE_CATALOG_VALID_JSON = "cb-image-catalog-valid.json";
+
+    private static final String CB_IMAGE_CATALOG_FILTER_NULL_IMAGES_JSON = "cb-image-catalog-filter-null-images.json";
+
+    private static final String CB_IMAGE_CATALOG_FILTER_IMAGES_NULL_JSON = "cb-image-catalog-filter-images-null.json";
+
+    private static final String FREEIPA_IMAGE_CATALOG_FILTER_IMAGES_NULL_JSON = "freeipa-image-catalog-filter-images-null.json";
+
+    private static final String CB_IMAGE_CATALOG_EMPTY_CLOUDBREAK_VERSIONS_JSON = "cb-image-catalog-empty-cloudbreak-versions.json";
+
+    private static final String CB_IMAGE_CATALOG_WITHOUT_BASE_IMAGES = "cb-image-catalog-without-base-images.json";
+
+    private static final String CB_IMAGE_CATALOG_WITHOUT_CDH_IMAGES = "cb-image-catalog-without-cdh-images.json";
+
+    private static final String CB_VERSION = "1.16.5";
+
+    private static final List<String> RC_IMAGE_CATALOG_OS_TYPES = Lists.newArrayList("amazonlinux", "centos7", "amazonlinux2", "sles12", "ubuntu16");
+
+    private static final List<String> CB_AMAZONLINUX_FILTER = Lists.newArrayList("amazonlinux");
+
+    private static final List<String> CB_REDHAT7_FILTER = Lists.newArrayList("redhat7");
+
+    @InjectMocks
+    private CachedImageCatalogWrapperProvider underTest;
+
+    @Mock
+    private Client clientMock;
+
+    @Mock
+    private WebTarget webTargetMock;
+
+    @Mock
+    private Invocation.Builder builderMock;
+
+    @Mock
+    private Response responseMock;
+
+    @Mock
+    private Response.StatusType statusTypeMock;
+
+    @Mock
+    private ImageCatalogServiceProxy imageCatalogServiceProxy;
+
+    @Spy
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private Map<ConfigKey, Client> clientMap = new HashMap<>();
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        clientMap.put(new ConfigKey(false, false, false), clientMock);
+
+        Field field = RestClientUtil.class.getDeclaredField("CLIENTS");
+        field.setAccessible(true);
+
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+
+        field.set(null, clientMap);
+    }
+
+    @Test
+    public void testReadImageCatalogFromFile() throws Exception {
+
+        String path = getPath(CB_IMAGE_CATALOG_V2_JSON);
+        ReflectionTestUtils.setField(underTest, "etcConfigDir", path);
+        ReflectionTestUtils.setField(underTest, "enabledLinuxTypes", Collections.emptyList());
+
+        CloudbreakImageCatalogV3 catalog = underTest.getImageCatalogWrapper(CB_IMAGE_CATALOG_V2_JSON).getImageCatalog();
+
+        assertNotNull("Check that the parsed ImageCatalog not null.", catalog);
```

Comment:
Can you please use `org.junit.jupiter.api.Assertions.assertNotNull` instead of `org.junit.Assert.assertNotNull`?

Reply:
It was not me as it is a refactored code, but sure, I can replace the questionable implementation.